internal/statushistory/statushistory.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package statushistory
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/juju/juju/core/logger"
+	"github.com/juju/juju/core/status"
+)
+
+// Recorder is an interface for recording status information.
+type Recorder interface {
+	// Logf logs information at the given level.
+	// The provided arguments are assembled together into a string with
+	// fmt.Sprintf.
+	Logf(ctx context.Context, level logger.Level, labels logger.Labels, format string, args ...any)
```

Comment:
This seems to break encapsulation. The Recorder interface should be backend agnostic and we then have a Logger implementation of it, the construction of which takes a Logf implementation. 
The fact that this method doesn't return an error also seems problematic - when loki is plugged in as the logger backend, we'd at least want to be aware an error has occurred and perhaps fall back to logging that locally. 

In terms of the labels, I think we're missing a "category", which should be "status_history". There will be other categorised temporal data streams we shove into logs and besides the domain and entity id, we'll also want to know what kind or category of data it it. I thought there was a spec but can't find it now.

Reply:
Yeah, not sure what to do here. We either remove it or just call it logger?

> The fact that this method doesn't return an error also seems problematic

We can change that as the need arises. No need to lie.

> In terms of the labels, I think we're missing a "category", which should be "status_history".

Perfect 👍